gateway/src/main/java/com/microsoft/azure/cosmosdb/rx/internal/query/OffsetContinuationToken.java

```
@@ -0,0 +1,70 @@
+package com.microsoft.azure.cosmosdb.rx.internal.query;
+
+import com.microsoft.azure.cosmosdb.JsonSerializable;
+import com.microsoft.azure.cosmosdb.rx.internal.Utils;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class OffsetContinuationToken extends JsonSerializable {
+    private static final String TOKEN_PROPERTY_NAME = "sourceToken";
+    private static final String OFFSET_PROPERTY_NAME = "offset";
+    private static final Logger logger = LoggerFactory.getLogger(CompositeContinuationToken.class);
+
+    public OffsetContinuationToken(int offset, String sourceToken) {
+
+        if (offset < 0) {
+            throw new IllegalArgumentException("offset should be non negative");
+        }
+
+        this.setOffset(offset);
+        this.setSourceToken(sourceToken);
+    }
+
+    public OffsetContinuationToken(String serializedCompositeToken) {
+        super(serializedCompositeToken);
+        this.getOffset();
+        this.getSourceToken();
+    }
+
+    private void setSourceToken(String sourceToken) {
+        super.set(TOKEN_PROPERTY_NAME, sourceToken);
+    }
+
+    private void setOffset(int offset) {
+        super.set(OFFSET_PROPERTY_NAME, offset);
+    }
+
+    public String getSourceToken() {
+        return super.getString(TOKEN_PROPERTY_NAME);
+    }
+
+    public int getOffset() {
+        if (super.getInt(OFFSET_PROPERTY_NAME) == null) {
```

If the int is null then the token is probably malformed and you should throw an exception instead.

---

Yes propagating the exception now so that exception is thrown in tryParse